### PR TITLE
Implementing the new '@rocket.chat/ui-kit' UI components

### PR DESCRIPTION
**Pull request by haemza30**
This pull request includes several changes to Apps.ClickUp application. Here is a summary of changes made :

1. Updated the rocket.chat/apps-engine to "v1.37.0".
2. Implemented the new "@rocket.chat/ui-kit" block classes, which will allow for greater customization of the application's UI.
3. Replaced all instances of "openModalView" with "openSurfaceView" to maintain consistency with the new naming convention.
4. Replaced all instances of "updateModalView" with "updateSurfaceView" to maintain consistency with the new naming convention.
5. Implemented a custom "blockBuilder" method for creating blocks and block elements.
6. Fixed linting.

### CODE REVIEW

**Comment by mustafahasankhan** [comment]
Woah! Thanks for this, amazed to see such grand contribution.
Can you please resolve the merge conflicts and also share a video recording?

File: ClickUpApp.ts [hunk 1]
```diff
@@ -24,123 +15,90 @@ import { HttpStatusCode } from '@rocket.chat/apps-engine/definition/accessors';
...
+
+  private async autorizationCallback(token: IAuthData, user: IUser, read: IRead, modify: IModify, http: IHttp, persistence: IPersistence) {
+    if (token) {
+      const headers = {
+        Authorization: `${token?.token}`,
+      };
+
+      const userData = await http.get(`https://api.clickup.com/api/v2/user`, { headers });
+      if (userData.statusCode == HttpStatusCode.OK) {
+        await connect_user_to_clickup_uid(read, persistence, userData.data.user.id, user.id);
```
Comment (mustafahasankhan) [inline ClickUpApp.ts]: Please change this to persists userasync

**Review by mustafahasankhan — CHANGES_REQUESTED** [review]
Please see the comment and submit the video recording.

**Comment by haemza30** [comment]
I have made the requested changes and Fixed minor issues.

Below are the video recordings 

1. Importing workspaces from ClickUp:
[Screencast from 10-04-23 03:41:29 PM IST.webm](https://user-images.githubusercontent.com/85013735/230882528-3135207b-268f-42ef-b7f8-68b1fa3f4598.webm)

2. Task Management
[Screencast from 10-04-23 02:58:45 PM IST.webm](https://user-images.githubusercontent.com/85013735/230882567-6b55c52a-3108-45f8-81d2-f30823c5f141.webm)

File: src/lib/const.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export const getSpacesOfUrl = (workspaceId: string) => {
 };
 
 export const getFoldersOfUrl = (spaceId: string) => {
-    return `${ApiBaseUrl}/${ApiVersion.V2}/${ApiEndpoint.Space}/${ApiEndpoint.FoldersOf(spaceId)}`;
+    return `${ApiBaseUrl}/${ApiVersion.V2}/space/${ApiEndpoint.FoldersOf(spaceId)}`;
```
Comment (mustafahasankhan) [inline src/lib/const.ts:53]: Please change space to ${ApiEndpoint.Space}

**Comment by mustafahasankhan** [comment]
LGTM, thanks!

**Pull request merged**